src/collective/workspace/membership.py

```diff
@@ -90,18 +90,25 @@ def _update_groups(self, old_groups, new_groups, add_auto_groups=True):
         context = workspace.context
         uid = context.UID()
         workspace_groups = get_workspace_groups_plugin()
-
         if self.user is None:
             return
 
         # Determine automatic groups
-        for name, condition in workspace.auto_groups.items():
-            if name not in workspace.available_groups:
-                raise Exception('Unknown workspace group: {}'.format(name))
-            if add_auto_groups and condition(self):
-                new_groups = new_groups.copy() | set([name])
-            else:
-                old_groups = old_groups.copy() | set([name])
+        if add_auto_groups:
+            new_groups = set(new_groups)
+            old_groups = set(old_groups)
+            for name, condition in workspace.auto_groups.items():
+                if name not in workspace.available_groups:
+                    raise Exception('Unknown workspace group: {}'.format(name))
+                if name in new_groups or name in old_groups:
+                    # Respect user input
+                    pass
```

Comment:
What is this for? Automatic groups are meant to be determined automatically based on the condition, not based on user input.

Reply:
Well, I started working on this after the issue #27, where in the migration we wanted to update the groups to include the user in group "Members" but given the existent code this was not happening.
Then I wrote the tests (based on what I thought it should happen when calling this function) and started to make the code behave as I think it should.

If needed I can remove this line, but it is giving the developer a way to override part of the automatic groups.

Reply:
I commented on #27 with what I think we should do for that issue. I think we should NOT force adding Members there, but let the auto groups get processed normally. The `add_auto_groups=False` option is really meant for one purpose, which is to make sure the user is removed from all groups when the membership is being removed from the roster.

I don't think it's a good idea to let the auto group conditions be overridden. Or at least I'd like to have a real use case in mind in order to consider it.

Reply:
Oh, I see, I will comment there.